### PR TITLE
Add missing TeamPolicyInternal<OpenMPTarget> converting constructor

### PR DESCRIPTION
**Pull request by dalg24**
Realized it was missing when working on #3379

### CODE REVIEW

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp [hunk 1]
```diff
@@ -813,8 +813,8 @@ class TeamPolicyInternal<Kokkos::Experimental::OpenMPTarget, Properties...>
   int m_vector_length;
   int m_team_alloc;
   int m_team_iter;
-  size_t m_team_scratch_size[2];
-  size_t m_thread_scratch_size[2];
+  std::array<size_t, 2> m_team_scratch_size;
```
Comment (crtrott) [inline core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp:816]: please don't I am not confident about std::array in all the OpenMPTarget compilers we try to use.

**Review by crtrott — CHANGES_REQUESTED** [review]
I am not confident about the std::array change.

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp (hunk 1 above)
Comment (crtrott) [reply core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp:816]: nvm in OpenMPTarget we don't map the policy.

**Review by crtrott — APPROVED** [review]

**Review by rgayatri23 — APPROVED** [review]
> I am not confident about the std::array change.

I share the same concern  but other than that LGTM.

File: core/unit_test/TestPolicyConstruction.hpp [hunk 2]
```diff
@@ -699,4 +699,21 @@ TEST(TEST_CATEGORY, policy_construction) {
   TestTeamPolicyConstruction<TEST_EXECSPACE>();
 }
 
+template <template <class...> class Policy, class... Args>
+void check_converting_constructor_add_work_tag(Policy<Args...> const& policy) {
+  // Not the greatest but at least checking it compiles
```
[the remark this answers is not in the document]
Comment (dalg24) [reply core/unit_test/TestPolicyConstruction.hpp:704]: I am not sure what we can do unless we provide `operator==`

**Pull request merged**